server/src/main/java/org/opensearch/node/Node.java

```diff
@@ -662,6 +667,17 @@ protected Node(
                 repositoriesServiceReference::get
             );
 
+            final BiFunction<IndexSettings, ShardRouting, TranslogFactory> translogFactorySupplier = (indexSettings, shardRouting) -> {
```

Comment:
Sort of a question of style, but I generally prefer creating a new functional interface type with a helpful name when you get signatures like `BiFunction<IndexSettings, ShardRouting, TranslogFactory>`. It will be more concise. The thing is this would end up named something like "TranslogFactorySupplier" which is basically a synonym for "TranslogFactoryFactory". What do you think? Is there a better name or is "TranslogFactorySupplier" okay? Looking briefly I don't see any obvious way to refactor to avoid this new type, but maybe there's another way.

Reply:
I had thought of creating a FunctionalInterface and tried seeing what is the status quo we have in opensearch. I could see that we have @FunctionalInterface interfaces as well as BiFunctions used across. I also stumbled upon https://github.com/opensearch-project/OpenSearch/tree/main/server/src/main/java/org/opensearch/common where we have classes like [AsyncBiFunction](https://github.com/opensearch-project/OpenSearch/blob/main/server/src/main/java/org/opensearch/common/AsyncBiFunction.java), [TriFunction](https://github.com/opensearch-project/OpenSearch/blob/main/server/src/main/java/org/opensearch/common/TriFunction.java). 

I believed that with BiFunction we will keep the same style and avoid creating another interface/class. I am good with either approach. Is there a third way? Probably / probably not. 

On the naming, I guess any other alternate would have the same synonym.

Reply:
Why do we need this to be far up in `Node`?

Reply:
There was a point raised about having the translogFactorySupplier as a singleton by @sachinpkale. Since this does not have any references to Shard specific details and to keep it singleton, we can either keep it here or in IndicesService. Any suggestions?

Reply:
As discussed offline @Bukhtawar, I have pushed down this to `IndicesService`class.